### ISSUE TITLE
Create (or Find) a guide for creating notices for communicating changes

### ISSUE THREAD

**Issue by dstufft** [pypi/warehouse#11805]
In the effort to move the top N projects to be secured using 2FA, we sent an email out to all of those authors. There was some obvious confusion with that email where people believed that this email was signaling that 2FA was being required *today*, rather than notifying them that it was going to be required in the future.

This isn't the first time that we've made a similar style change, and the emails have caused confused with people.

It would be great if we could either find or create a guide to what these kinds of emails should have, maybe in a check list form, so that we can reference it in the future. Maybe we could even have a standard "email campaign" base template that implements that that has call outs for things like "what is changing today", "what is changing in the future", "what does this mean for me", etc.

I'm not 100% sure what this would look like, but it may make these kinds of things go smoother in the future.

**Comment by pradyunsg** [comment]
I think useful headings to have are:

- What you need to know
- What is changing today
- What will change in the future
- What you need to do
- What you can do
- Why this is being done
- How does this affect you

We don't need to use all these heading, nor only these headings, nor in this order -- but the general style that they follow makes things clearer IMO. I think trying to summarize "things to know" under the first heading of the email is a good idea. In case of this email, that would have been something like:

> # What you need to know
>
> One or more of your projects have been identified as "critical". This is because it is in the top 1% of packages, based on the download statistics. In the near future, managing this project will require you to have 2-factor authentication enabled on your PyPI account.

Ideally, we'd have had a date to share here, instead of saying "near future".

Personally, https://pycon.blogspot.com/2022/04/important-venue-information-update.html is a good example of clearly communicating a bunch of information to the reader, while keeping it efficient and making to identify actionable steps.

Beynd that, as a matter of process, I think having a test run sending emails only to the moderators and admins, would allow for an early sweep to catch a certain class of issues as well (such as the localhost link being sent, instead of the pypi.org link).

Beyond that:

- I think PyPI communication should take a neutral tone in all the emails -- similar to what we use in most of the status updates over on status.python.org.
- It's vital to make it clear whether the user needs to take action, or whether this is a notification to inform them of an action, or if this is an informational message about something that will affect them.

**Comment by dstufft** [comment]
Talking to someone about this, and they pointed out that probably the biggest thing we could do here is try to get our comms reviewed by people who are somewhat divorced from the implementation, particularly people of different backgrounds (non native english speakers, etc) to review our comms prior to them going out to see how people who aren't quite as close to the change or following exactly what's happening all of the time interpret our messaging.

Not sure what the best way of handling that is. We don't need to do this kind of communication *that* often, but this isn't hardly the first time (nor is it likely to be the last), and every time we've done it, I think we've had some level of blowback that could be attributed to people reading our comms with different ideas of what it meant than we had.

Could be as simple as making a post on discuss and manually shopping it around to people, or a mailing list or something like that. I dunno, but it's probably a good idea I think.

**Comment by glyph** [comment]
Another thing that I think might be useful to consider here is timing. I wasn't closely involved with the 2FA thing but I was excited for it, but upon reflection one of the big reasons was that it was not a *surprise*.  Telegraphing changes — particularly changes that might read as "mandates" (like the non-optionalness of the "critical" designation) — with as much time as possible will give people time to understand what is happening and why, and get comfortable with it before it actually happens.

**Comment by glyph** [comment]
(It wasn't a surprise *to me*, I mean, but that was an accident of history; just random that I happen to be plugged in socially to this sort of stuff.)

**Comment by dstufft** [comment]
Well, nothing has changed on PyPI yet, the notice that went out the other day was intended to be that notice, that we're going to require it at some point in the future, but nothing was changing for them yet.

Although maybe step 1 before emails, should have been a blog post or something as a more general notice that things were going to happen, then the emails go out to say "hey things are going to happen, and also you're directly affected",then finally at some point the thing actually happens.

**Comment by glyph** [comment]
> Well, nothing has changed on PyPI yet, the notice that went out the other day was intended to be that notice, that we're going to require it at some point in the future, but nothing was changing for them yet.
> 
> Although maybe step 1 before emails, should have been a blog post or something as a more general notice that things were going to happen, then the emails go out to say "hey things are going to happen, and also you're directly affected",then finally at some point the thing actually happens.

Exactly. Something *did* change (the criticality designation in the UI), although the actually impactful change (the 2FA requirement) hasn't actually rolled out yet.  But the presence of the visible change I think confused a lot of people as to what exactly was changing and when.

The email I am thinking about is one which will be very explicitly "*nothing* has changed yet, here are when things *will* change".
